### PR TITLE
Fix 'Not found column' error in MaterializeMySQL

### LINKED ISSUE #23708: Another case encountered "DB::Exception: Not found column xxx in block..."  (MaterializeMySQL)
I reported the issue #23470 a few days ago, now it is fine after upgrade to the latest stable version (version 21.4.5.46).
But encountered same error message "`Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Not found column type in block: while executing 'INPUT : 1 -> type Nullable(Int8) : 1'`" for another case...

Refer to below two screenshots, it's fine if the column **type** exits in both 'select' and 'where' clause.
image
image

but encountered the exception if the column **type** only exists in 'where' clause:
image

### PR DESCRIPTION
**Pull request by tavplubix**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed `Not found column ...` and `Missing column ...` errors when selecting from `MaterializeMySQL`. Fixes #23708, #24830, #25794

### CODE REVIEW

File: src/Parsers/ASTSelectQuery.cpp [hunk 1]
```diff
@@ -438,4 +438,19 @@ ASTPtr & ASTSelectQuery::getExpression(Expression expr)
...
 }
 
+void ASTSelectQuery::setFinal()
+{
+    auto & tables_in_select_query = tables()->as<ASTTablesInSelectQuery &>();
+
+    if (tables_in_select_query.children.empty())
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Tables list is empty, it's a bug");
+
+    auto & tables_element = tables_in_select_query.children[0]->as<ASTTablesInSelectQueryElement &>();
```
Comment (azat) [inline src/Parsers/ASTSelectQuery.cpp:448]: But is it OK to set FINAL only for one table?

Comment (tavplubix) [reply src/Parsers/ASTSelectQuery.cpp:448]: Yes, it's checked for the first table only:
https://github.com/ClickHouse/ClickHouse/blob/fd914f3d9715df06bef5611591b7852b8a9dc66c/src/Parsers/ASTSelectQuery.cpp#L299-L306

**Comment by nikitamikhaylov** [comment]
@Mergifyio update

**Comment by nikitamikhaylov** [comment]
@Mergifyio update

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**